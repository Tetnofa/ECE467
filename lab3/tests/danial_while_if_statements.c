int main() {
    int x;
    x = 1;
    if (x == 1) {
        if (x == 1) {
            int y;
            y = 1;
            if (y != 1) {

            } else {
                if (y <= 2) {
                    if (z) {
                        if (z) {

                        }
                    }
                }
            }
        }
    }
}
void poo () {
    if (1 < x) {

    }
}
int a (int b, int c) {
    if (b < c) {
        if (x) {

        }else {
            if (b != c) {

            }
        }
    }
}
int f (int arr[],int b, int c) {
    if (arr[3] <= b) {
        f(arr, b, c);
    } else if (arr[3] >= c) {
        f(arr, b, c);
    } else if (arr[5] > y) {
        poo();
    }
}
void fcn() {
    while (1 < 2) {

    }
    while (x) {

    }
    while (y) {
        while ( 1 < 2) {
            while (x) {
                if (f) {
                    if (1 == 2) {
                        while (1 != 8) {
                            if (5==3) {
                                if (5 <= 3) {
                                    while (3) {

                                    }
                                }
                            }
                        }
                    }
                }
            }
        }
    }
}

void whileScope() {
    while (1) {
        int x;
        x = 2;
        if (x == 2) {
            int y;
        }
    }
}
void pee() {
    int x;
    while (1) {
        int y;
        y = 9;
        if (x <= y) {
            if (x >= 9) {
                if (x != y) {

                }
            }
        }
    }
}
bool fadwa() {
    bool x;
    bool y;
    while (x) {
        if (y) {

        }
    }
}