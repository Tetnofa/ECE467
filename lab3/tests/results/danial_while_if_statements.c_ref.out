Program [useIO=0](1,0)
	Function[isProto=0]	Int(1,0)
		Identifier[name:main](1,4)
		Scope(1,11)
			Scalar Declaration			Int(2,4)
				Identifier[name:x](2,8)
			Assignment(3,4)
				Reference(3,4)
					Identifier[name:x](3,4)
				Int Expression(3,8)
					IntConstant[val=1](3,8)
			IfStmt[hasElse=0](4,4)
				Bool Expression(4,8)
					Binary Expression[opcode:==](4,8)
						Int Expression(4,8)
							Reference(4,8)
								Identifier[name:x](4,8)
						Int Expression(4,13)
							IntConstant[val=1](4,13)
				Scope(4,16)
					IfStmt[hasElse=0](5,8)
						Bool Expression(5,12)
							Binary Expression[opcode:==](5,12)
								Int Expression(5,12)
									Reference(5,12)
										Identifier[name:x](5,12)
								Int Expression(5,17)
									IntConstant[val=1](5,17)
						Scope(5,20)
							Scalar Declaration							Int(6,12)
								Identifier[name:y](6,16)
							Assignment(7,12)
								Reference(7,12)
									Identifier[name:y](7,12)
								Int Expression(7,16)
									IntConstant[val=1](7,16)
							IfStmt[hasElse=1](8,12)
								Bool Expression(8,16)
									Binary Expression[opcode:!=](8,16)
										Int Expression(8,16)
											Reference(8,16)
												Identifier[name:y](8,16)
										Int Expression(8,21)
											IntConstant[val=1](8,21)
								Scope(8,24)
								Scope(10,19)
									IfStmt[hasElse=0](11,16)
										Bool Expression(11,20)
											Binary Expression[opcode:<=](11,20)
												Int Expression(11,20)
													Reference(11,20)
														Identifier[name:y](11,20)
												Int Expression(11,25)
													IntConstant[val=2](11,25)
										Scope(11,28)
											IfStmt[hasElse=0](12,20)
												Int Expression(12,24)
													Reference(12,24)
														Identifier[name:z](12,24)
												Scope(12,27)
													IfStmt[hasElse=0](13,24)
														Int Expression(13,28)
															Reference(13,28)
																Identifier[name:z](13,28)
														Scope(13,31)
	Function[isProto=0]	Void(22,0)
		Identifier[name:poo](22,5)
		Scope(22,12)
			IfStmt[hasElse=0](23,4)
				Bool Expression(23,8)
					Binary Expression[opcode:<](23,8)
						Int Expression(23,8)
							IntConstant[val=1](23,8)
						Int Expression(23,12)
							Reference(23,12)
								Identifier[name:x](23,12)
				Scope(23,15)
	Function[isProto=0]	Int(27,0)
		Identifier[name:a](27,4)
		Parameter		Int(27,7)
			Identifier[name:b](27,11)
		Parameter		Int(27,14)
			Identifier[name:c](27,18)
		Scope(27,21)
			IfStmt[hasElse=0](28,4)
				Bool Expression(28,8)
					Binary Expression[opcode:<](28,8)
						Int Expression(28,8)
							Reference(28,8)
								Identifier[name:b](28,8)
						Int Expression(28,12)
							Reference(28,12)
								Identifier[name:c](28,12)
				Scope(28,15)
					IfStmt[hasElse=1](29,8)
						Int Expression(29,12)
							Reference(29,12)
								Identifier[name:x](29,12)
						Scope(29,15)
						Scope(31,14)
							IfStmt[hasElse=0](32,12)
								Bool Expression(32,16)
									Binary Expression[opcode:!=](32,16)
										Int Expression(32,16)
											Reference(32,16)
												Identifier[name:b](32,16)
										Int Expression(32,21)
											Reference(32,21)
												Identifier[name:c](32,21)
								Scope(32,24)
	Function[isProto=0]	Int(38,0)
		Identifier[name:f](38,4)
		Parameter		Int*(38,7)
			Identifier[name:arr](38,11)
		Parameter		Int(38,17)
			Identifier[name:b](38,21)
		Parameter		Int(38,24)
			Identifier[name:c](38,28)
		Scope(38,31)
			IfStmt[hasElse=1](39,4)
				Bool Expression(39,8)
					Binary Expression[opcode:<=](39,8)
						Int Expression(39,8)
							Reference(39,8)
								Identifier[name:arr](39,8)
								Int Expression(39,12)
									IntConstant[val=3](39,12)
						Int Expression(39,18)
							Reference(39,18)
								Identifier[name:b](39,18)
				Scope(39,21)
					Expression Statement(40,8)
						CallExpr(40,8)
							Identifier[name:f](40,8)
							Argument(40,10)
								Int Expression(40,10)
									Reference(40,10)
										Identifier[name:arr](40,10)
							Argument(40,15)
								Int Expression(40,15)
									Reference(40,15)
										Identifier[name:b](40,15)
							Argument(40,18)
								Int Expression(40,18)
									Reference(40,18)
										Identifier[name:c](40,18)
				IfStmt[hasElse=1](41,11)
					Bool Expression(41,15)
						Binary Expression[opcode:>=](41,15)
							Int Expression(41,15)
								Reference(41,15)
									Identifier[name:arr](41,15)
									Int Expression(41,19)
										IntConstant[val=3](41,19)
							Int Expression(41,25)
								Reference(41,25)
									Identifier[name:c](41,25)
					Scope(41,28)
						Expression Statement(42,8)
							CallExpr(42,8)
								Identifier[name:f](42,8)
								Argument(42,10)
									Int Expression(42,10)
										Reference(42,10)
											Identifier[name:arr](42,10)
								Argument(42,15)
									Int Expression(42,15)
										Reference(42,15)
											Identifier[name:b](42,15)
								Argument(42,18)
									Int Expression(42,18)
										Reference(42,18)
											Identifier[name:c](42,18)
					IfStmt[hasElse=0](43,11)
						Bool Expression(43,15)
							Binary Expression[opcode:>](43,15)
								Int Expression(43,15)
									Reference(43,15)
										Identifier[name:arr](43,15)
										Int Expression(43,19)
											IntConstant[val=5](43,19)
								Int Expression(43,24)
									Reference(43,24)
										Identifier[name:y](43,24)
						Scope(43,27)
							Expression Statement(44,8)
								CallExpr(44,8)
									Identifier[name:poo](44,8)
	Function[isProto=0]	Void(47,0)
		Identifier[name:fcn](47,5)
		Scope(47,11)
			WhileStmt(48,4)
				Bool Expression(48,11)
					Binary Expression[opcode:<](48,11)
						Int Expression(48,11)
							IntConstant[val=1](48,11)
						Int Expression(48,15)
							IntConstant[val=2](48,15)
				Scope(48,18)
			WhileStmt(51,4)
				Int Expression(51,11)
					Reference(51,11)
						Identifier[name:x](51,11)
				Scope(51,14)
			WhileStmt(54,4)
				Int Expression(54,11)
					Reference(54,11)
						Identifier[name:y](54,11)
				Scope(54,14)
					WhileStmt(55,8)
						Bool Expression(55,16)
							Binary Expression[opcode:<](55,16)
								Int Expression(55,16)
									IntConstant[val=1](55,16)
								Int Expression(55,20)
									IntConstant[val=2](55,20)
						Scope(55,23)
							WhileStmt(56,12)
								Int Expression(56,19)
									Reference(56,19)
										Identifier[name:x](56,19)
								Scope(56,22)
									IfStmt[hasElse=0](57,16)
										Int Expression(57,20)
											Reference(57,20)
												Identifier[name:f](57,20)
										Scope(57,23)
											IfStmt[hasElse=0](58,20)
												Bool Expression(58,24)
													Binary Expression[opcode:==](58,24)
														Int Expression(58,24)
															IntConstant[val=1](58,24)
														Int Expression(58,29)
															IntConstant[val=2](58,29)
												Scope(58,32)
													WhileStmt(59,24)
														Bool Expression(59,31)
															Binary Expression[opcode:!=](59,31)
																Int Expression(59,31)
																	IntConstant[val=1](59,31)
																Int Expression(59,36)
																	IntConstant[val=8](59,36)
														Scope(59,39)
															IfStmt[hasElse=0](60,28)
																Bool Expression(60,32)
																	Binary Expression[opcode:==](60,32)
																		Int Expression(60,32)
																			IntConstant[val=5](60,32)
																		Int Expression(60,35)
																			IntConstant[val=3](60,35)
																Scope(60,38)
																	IfStmt[hasElse=0](61,32)
																		Bool Expression(61,36)
																			Binary Expression[opcode:<=](61,36)
																				Int Expression(61,36)
																					IntConstant[val=5](61,36)
																				Int Expression(61,41)
																					IntConstant[val=3](61,41)
																		Scope(61,44)
																			WhileStmt(62,36)
																				Int Expression(62,43)
																					IntConstant[val=3](62,43)
																				Scope(62,46)
	Function[isProto=0]	Void(75,0)
		Identifier[name:whileScope](75,5)
		Scope(75,18)
			WhileStmt(76,4)
				Int Expression(76,11)
					IntConstant[val=1](76,11)
				Scope(76,14)
					Scalar Declaration					Int(77,8)
						Identifier[name:x](77,12)
					Assignment(78,8)
						Reference(78,8)
							Identifier[name:x](78,8)
						Int Expression(78,12)
							IntConstant[val=2](78,12)
					IfStmt[hasElse=0](79,8)
						Bool Expression(79,12)
							Binary Expression[opcode:==](79,12)
								Int Expression(79,12)
									Reference(79,12)
										Identifier[name:x](79,12)
								Int Expression(79,17)
									IntConstant[val=2](79,17)
						Scope(79,20)
							Scalar Declaration							Int(80,12)
								Identifier[name:y](80,16)
	Function[isProto=0]	Void(84,0)
		Identifier[name:pee](84,5)
		Scope(84,11)
			Scalar Declaration			Int(85,4)
				Identifier[name:x](85,8)
			WhileStmt(86,4)
				Int Expression(86,11)
					IntConstant[val=1](86,11)
				Scope(86,14)
					Scalar Declaration					Int(87,8)
						Identifier[name:y](87,12)
					Assignment(88,8)
						Reference(88,8)
							Identifier[name:y](88,8)
						Int Expression(88,12)
							IntConstant[val=9](88,12)
					IfStmt[hasElse=0](89,8)
						Bool Expression(89,12)
							Binary Expression[opcode:<=](89,12)
								Int Expression(89,12)
									Reference(89,12)
										Identifier[name:x](89,12)
								Int Expression(89,17)
									Reference(89,17)
										Identifier[name:y](89,17)
						Scope(89,20)
							IfStmt[hasElse=0](90,12)
								Bool Expression(90,16)
									Binary Expression[opcode:>=](90,16)
										Int Expression(90,16)
											Reference(90,16)
												Identifier[name:x](90,16)
										Int Expression(90,21)
											IntConstant[val=9](90,21)
								Scope(90,24)
									IfStmt[hasElse=0](91,16)
										Bool Expression(91,20)
											Binary Expression[opcode:!=](91,20)
												Int Expression(91,20)
													Reference(91,20)
														Identifier[name:x](91,20)
												Int Expression(91,25)
													Reference(91,25)
														Identifier[name:y](91,25)
										Scope(91,28)
	Function[isProto=0]	Bool(98,0)
		Identifier[name:fadwa](98,5)
		Scope(98,13)
			Scalar Declaration			Bool(99,4)
				Identifier[name:x](99,9)
			Scalar Declaration			Bool(100,4)
				Identifier[name:y](100,9)
			WhileStmt(101,4)
				Int Expression(101,11)
					Reference(101,11)
						Identifier[name:x](101,11)
				Scope(101,14)
					IfStmt[hasElse=0](102,8)
						Int Expression(102,12)
							Reference(102,12)
								Identifier[name:y](102,12)
						Scope(102,15)
sema: 12:24 : use of undefined identifier z
sema: 13:28 : use of undefined identifier z
sema: 13:24 : invalid condition in if statement
sema: 12:20 : invalid condition in if statement
sema: 23:12 : use of undefined identifier x
sema: 23:8 : type mismatch 
sema: 23:4 : invalid condition in if statement
sema: 29:12 : use of undefined identifier x
sema: 29:8 : invalid condition in if statement
sema: 43:24 : use of undefined identifier y
sema: 43:15 : type mismatch 
sema: 43:11 : invalid condition in if statement
sema: 51:11 : use of undefined identifier x
sema: 51:4 : invalid condition in while statement
sema: 54:11 : use of undefined identifier y
sema: 56:19 : use of undefined identifier x
sema: 57:20 : use of undefined identifier f
sema: 62:36 : invalid condition in while statement
sema: 57:16 : invalid condition in if statement
sema: 56:12 : invalid condition in while statement
sema: 54:4 : invalid condition in while statement
sema: 76:4 : invalid condition in while statement
sema: 86:4 : invalid condition in while statement
